scripts/fast-reboot

```diff
@@ -70,44 +88,83 @@ docker exec -i bgp killall -9 bgpd
 docker kill lldp > /dev/null
 
 # Kill teamd, otherwise it gets down all LAGs
+# Note: teamd must be killed before syncd, because it will send the last packet through CPU port
+# TODO: stop teamd gracefully to allow teamd to send last valid update to be sure we'll have 90 seconds reboot time
 docker kill teamd > /dev/null
 
+if [[ "$REBOOT_TYPE" = "warm-reboot" ]]; then
+    # Freeze orchagent for warm restart
+    docker exec -i swss /usr/bin/orchagent_restart_check || {
+        echo "RESTARTCHECK failed\n" >&2
+        # TODO: remove this sleep if we have better timing control
+        sleep 1
+        exit 10
+    }
+    # Set whole system warm reboot flag
+    config warm_restart enable system
+fi
+# Kill swss dockers
+docker kill swss
+
 # syncd graceful stop is supported only for Broadcom platforms only for now
-if [[ "$sonic_asic_type" = 'broadcom' ]];
-then
+if [[ "$sonic_asic_type" = 'broadcom' ]]; then
     # Gracefully stop syncd
-    docker exec -i syncd /usr/bin/syncd_request_shutdown --cold > /dev/null
+    if [[ "$REBOOT_TYPE" = "fast-reboot" ]]; then
+        docker exec -i syncd /usr/bin/syncd_request_shutdown --cold > /dev/null
+    elif [[ "$REBOOT_TYPE" = "warm-reboot" ]]; then
+        docker exec -i syncd /usr/bin/syncd_request_shutdown --warm > /dev/null
+    fi
```

Comment:
Alternatively, in warm-reboot branch, you can call "systemctrl stop syncd" to stop syncd service, it has proper warm reboot support. If you do this, you could also remove "docker kill syncd" below.,

Reply:
I cannot find the code you mentioned "it has proper warm reboot support". Could you give a link?

---
In reply to: [225445275](https://github.com/Azure/sonic-utilities/pull/338#discussion_r225445275) [](ancestors = 225445275)

Reply:
Actually it is good to keep it in if-else block because it is a special case for syncd_request_shutdown.
You can see the mapping is weird
- fast-reboot -> cold
- warm-reboot -> warm
It doesn't make sense to reuse this logic in other places.

---
In reply to: [225442543](https://github.com/Azure/sonic-utilities/pull/338#discussion_r225442543) [](ancestors = 225442543)

Reply:
warm boot logic in syncd service: https://github.com/Azure/sonic-buildimage/blob/master/files/scripts/syncd.sh#L42

I disagree with keeping the logic in if-else block but this is a minor issue. I am ok with your decision.

Reply:
I got your idea, and fixed.

---
In reply to: [225674043](https://github.com/Azure/sonic-utilities/pull/338#discussion_r225674043) [](ancestors = 225674043,225445275)

Reply:
Fixed.

---
In reply to: [225726671](https://github.com/Azure/sonic-utilities/pull/338#discussion_r225726671) [](ancestors = 225726671)